http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -0,0 +1,205 @@
+          IO.raiseError(
+            new WorkbenchException(
+              s"Failed to process StopUpdateMessage for Cluster ${resolvedCluster.projectNameString}. This is likely due to a mismatch in state between the db and the message, or an improperly formatted machineConfig in the message. Cluster details: ${resolvedCluster}"
+            )
+          )
+        case None =>
+          IO.raiseError(
+            new WorkbenchException(
+              s"Could process StopUpdateMessage for cluster with id ${message.clusterId} because it was not found in the database"
+            )
+          )
+      }
+
+  def handleClusterTransitionFinished(message: ClusterTransitionFinishedMessage): IO[Unit] =
+    message.clusterFollowupDetails.clusterStatus match {
+      case Stopped => {
+        for {
+          clusterOpt <- dbRef.inTransaction { clusterQuery.getClusterById(message.clusterFollowupDetails.clusterId) }
+          savedMasterMachineType <- dbRef.inTransaction {
+            followupQuery.getFollowupAction(message.clusterFollowupDetails)
+          }
+          result <- clusterOpt match {
+            case Some(resolvedCluster) if resolvedCluster.status != ClusterStatus.Stopped =>
+              IO.raiseError(
+                new WorkbenchException(
+                  s"Unable to process message ${message} for cluster ${resolvedCluster.projectNameString} in status ${resolvedCluster.status.toString}, when the monitor signalled it stopped as it is not stopped."
+                )
+              )
+
+            case Some(resolvedCluster) => {
+
+              savedMasterMachineType match {
+                case Some(machineType) =>
+                  for {
+                    // perform gddao and db updates for new resources
+                    _ <- clusterHelper.updateMasterMachineType(resolvedCluster, MachineType(machineType))
+                    // start cluster
+                    _ <- clusterHelper.internalStartCluster(resolvedCluster)
+                    // clean-up info from follow-up table
+                    _ <- dbRef.inTransaction { followupQuery.delete(message.clusterFollowupDetails) }
+                  } yield ()
+                case None => IO.unit //the database has no record of a follow-up being needed. This is a no-op
+              }
+            }
+
+            case None => IO.raiseError(ClusterNotFoundException(message.clusterFollowupDetails.clusterId))
+          }
+        } yield result
+      }
+
+      //No actions for other statuses yet. There is some logic that will be needed for all other cases (i.e. the 'None' case where no cluster is found in the db and possibly the case that checks for the data in the DB)
+      // TODO: Refactor once there is more than one case
+      case _ => IO.unit
+    }
+
+}
+
+sealed trait LeoPubsubMessage {
+  def messageType: String
+}
+
+final case class StopUpdateMessage(updatedMachineConfig: MachineConfig, clusterId: Long) extends LeoPubsubMessage {
```

Comment:
can you move those sub classes of `LeoPubsubMessage` into an object `LeoPubsubMessage`?

like 
```
object LeoPubsubMessage {
 ...
}